samples: drivers: crypto: fix Mbed TLS shim driver testing

valeriosetti opened the pull request:
- Increase heap memory size
- Adjust regex matching so that it matches all the supported modes and it ensures that tests really passed

Tested with:
`twister -c -p native_sim -p qemu_riscv64 -p qemu_x86 -T samples/drivers/crypto/ -s sample.drivers.crypto.mbedtls`
and it's passing.

Fix #100572

@mars-low I cannot add you as reviewer, but since you created #100572 I would like to have a double-check confirmation from you that all tests are fine.

valeriosetti committed:
samples: drivers: crypto: fix heap memory size for Mbed TLS shim

Increase heap memory size when Mbed TLS shim driver is used otherwise
all crypto tests will fail on some platforms.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

valeriosetti committed:
samples: drivers: crypto: fix regex matching for Mbed TLS shim

Update regex matching for the Mbed TLS shim driver scenario in order to
match all the supported cases and ensure that they all passed.

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

mars-low commented:
Hi @valeriosetti , I responded in https://github.com/zephyrproject-rtos/zephyr/pull/100580#issuecomment-3630873233. Changes here look good as they fix qemu platforms and prevent silent failures, but the issue with misaligned access persists on `hifive_unmatched/fu740/s7` and `beaglev_fire/polarfire/u54`.

valeriosetti committed:
drivers: crypto: mbedtls_shim: fix crash in AES-ECB for RISCV 64 bits

Using "pkt->out_len" as the output length of "psa_cipher_[en|de]crypt"
caused a crash in RISCV 64 bits platforms due to misaligned address
access.
The solution is to add a temporary value on the stack to store this lenght
and only after the [en|de]cryption copy it to "pkt->out_len".

Signed-off-by: Valerio Setti <vsetti@baylibre.com>

fkokosinski reviewed (approved):
This fixes the issue for RISC-V platforms, thanks!

LGTM
